cardano-db-sync/src/Cardano/DbSync/Util.hs

```diff
@@ -185,15 +189,30 @@ whenMaybe :: Monad m => Maybe a -> (a -> m b) -> m (Maybe b)
 whenMaybe (Just a) f = Just <$> f a
 whenMaybe Nothing _f = pure Nothing
 
+foldAndAccM :: forall a b c t m. (Foldable t, Monad m) => ([c] -> a -> m (b, [c])) -> t a -> m [b]
+foldAndAccM f as = reverse . snd <$> foldM g ([], []) as
+  where
+    g :: ([c], [b]) -> a -> m ([c], [b])
+    g (cs, bs) a = do
+      (b, cs') <- f cs a
+      pure (cs <> cs', b : bs)
+
 third :: (a, b, c) -> c
 third (_, _, c) = c
 
-thrd3 :: (a, b, c, d) -> c
-thrd3 (_, _, c, _) = c
+first3 :: (a, b, c) -> a
+first3 (a, _, _) = a
```

Comment:
It looks like this is already provided by [extra](https://hackage.haskell.org/package/extra-1.7.14/docs/Data-Tuple-Extra.html#v:first3), which we already [use](https://github.com/IntersectMBO/cardano-db-sync/blob/master/cardano-db-sync/cardano-db-sync.cabal#L179)